src/PhpSpreadsheet/Worksheet/Table.php

```diff
@@ -0,0 +1,471 @@
+        }
+
+        $this->name = $name;
+
+        return $this;
+    }
+
+    /**
+     * Get show Header Row.
+     *
+     * @return bool
+     */
+    public function getShowHeaderRow()
+    {
+        return $this->showHeaderRow;
+    }
+
+    /**
+     * Set show Header Row.
+     *
+     * @return  $this
+     */
+    public function setShowHeaderRow(bool $showHeaderRow)
+    {
+        $this->showHeaderRow = $showHeaderRow;
+
+        return $this;
+    }
+
+    /**
+     * Get show Totals Row.
+     *
+     * @return bool
+     */
+    public function getShowTotalsRow()
+    {
+        return $this->showTotalsRow;
+    }
+
+    /**
+     * Set show Totals Row.
+     *
+     * @return  $this
+     */
+    public function setShowTotalsRow(bool $showTotalsRow)
+    {
+        $this->showTotalsRow = $showTotalsRow;
+
+        return $this;
+    }
+
+    /**
+     * Get Table Range.
+     *
+     * @return string
+     */
+    public function getRange()
+    {
+        return $this->range;
+    }
+
+    /**
+     * Set Table Cell Range.
+     *
+     * @return $this
+     */
+    public function setRange(string $range): self
+    {
+        // extract coordinate
+        [$worksheet, $range] = Worksheet::extractSheetTitle($range, true);
+        if (empty($range)) {
+            //    Discard all column rules
+            $this->columns = [];
+            $this->range = '';
+
+            return $this;
+        }
+
+        if (strpos($range, ':') === false) {
+            throw new PhpSpreadsheetException('Table must be set on a range of cells.');
+        }
+
+        $this->range = $range;
+        //    Discard any column ruless that are no longer valid within this range
+        [$rangeStart, $rangeEnd] = Coordinate::rangeBoundaries($this->range);
+        foreach ($this->columns as $key => $value) {
+            $colIndex = Coordinate::columnIndexFromString($key);
+            if (($rangeStart[0] > $colIndex) || ($rangeEnd[0] < $colIndex)) {
+                unset($this->columns[$key]);
+            }
+        }
+
+        return $this;
+    }
+
+    /**
+     * Set Table Cell Range to max row.
+     *
+     * @return $this
+     */
+    public function setRangeToMaxRow(): self
+    {
+        if ($this->workSheet !== null) {
+            $thisrange = $this->range;
+            $range = preg_replace('/\\d+$/', (string) $this->workSheet->getHighestRow(), $thisrange) ?? '';
+            if ($range !== $thisrange) {
+                $this->setRange($range);
+            }
+        }
+
+        return $this;
+    }
+
+    /**
+     * Get Table's Worksheet.
+     *
+     * @return null|Worksheet
+     */
+    public function getWorksheet()
+    {
+        return $this->workSheet;
+    }
+
+    /**
+     * Set Table's Worksheet.
+     *
+     * @return $this
+     */
+    public function setWorksheet(?Worksheet $worksheet = null)
+    {
+        if ($this->name != '' && $worksheet != null) {
```

Comment:
Strict comparisons would be better here than loose comparisons

Reply:
Changed wherever possible 3a6ebc0ce6de43c6f37c6331177767e12245b4e5